x/upgrade/types/plan.go

```diff
@@ -5,17 +5,31 @@ import (
 	"strings"
 	"time"
 
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+	ibcexported "github.com/cosmos/cosmos-sdk/x/ibc/exported"
 )
 
+var _ codectypes.UnpackInterfacesMessage = Plan{}
+
 func (p Plan) String() string {
 	due := p.DueAt()
 	dueUp := strings.ToUpper(due[0:1]) + due[1:]
+	var upgradedClientStr string
+	upgradedClient, err := clienttypes.UnpackClientState(p.UpgradedClientState)
```

Comment:
I'd suggest checking if `p.UpgradedClientState != nil` instead of allowing unpack with a nil client state. The err on unpacking could be return instead of `no upgraded client provided`

Reply:
Can't return an error because that would break `fmt.Stringer` interface

Reply:
why can't you set `upgradedClientStr = err.Error()`?